src/error.rs

```diff
@@ -0,0 +1,15 @@
+use std::fmt;
+
+#[derive(Debug, Clone)]
+pub struct DeError {
+    pub message: String,
+    pub line: usize,
+    pub column: usize,
+    pub jpart: String,
```

Comment:
I think if a `str` was given, `jpart` should have a pointer to that rather than creating a duplicate object. The memory usage could be meaningful. I'm inclined to leave off `jpart` because it's more involved than I have time to look at while line and column are obvious.